common/logger.py

```diff
@@ -55,14 +55,16 @@ def warning(msg , parent = None, traceDepth = 0):
 def info(msg , parent = None, traceDepth = 0):
     if DEBUG:
         msg = '%s %s' %(_debugHeader(parent, traceDepth), msg)
-    print('%sINFO%s: %s' %(bcolors.OKGREEN, bcolors.ENDC, msg), file=sys.stdout)
+    print('%sINFO%s: %s' %(bcolors.OKGREEN, bcolors.ENDC, msg), file=sys.stderr)
     for line in tools.wrapLine(msg):
         syslog.syslog(syslog.LOG_INFO, 'INFO: ' + line)
 
 def debug(msg, parent = None, traceDepth = 0):
     if DEBUG:
         msg = '%s %s' %(_debugHeader(parent, traceDepth), msg)
-        print('%sDEBUG%s: %s' %(bcolors.OKBLUE, bcolors.ENDC, msg), file = sys.stdout)
+        # Why does this code differ from eg. "error()"
```

Comment:
In `error()` an error message is written to syslog no matter if debug mode is on or not. But `if DEBUG` then debug infos are **added** to the error messgae.

In `debug()` there is no need to **add** debug infos to the message because the message itself is of type `DEBUG`.

Reply:
What is irritating to me is that DEBUG is **not** a severity threshold for logging but a flag to inject additional call stack information ("_debugHeader") but for a single severity level `debug()` it is also a threshold.
I do not know any logging framework that works like that (and don't understand why this makes sense).
Perhaps the idea was to work like `-v` vs. `-vv` verbosity switches (the more "v"s the chattier is the output ;-)

Reply:
Maybe we misunderstand here each other? e.g. in Python's own `logging` package `logging.DEBUG` is a threshold.
But you are deeper in that part of the code. It was just a quick and dirty night shift thought. :smile: